### PR TITLE
Update cloudnative owned packages to format_version 3.0.0

### PR DESCRIPTION
**Pull request by tommyers-elastic**
Update cloudnative packages to format_version 3.0.0

These changes are mostly autogenerated in two stages. 

1. Firstly the following `yq` commands are run across all the affected packages 

  ```
  find $1/data_stream -type f -path "*fields*" -name "*.yml" -exec yq e -i "del .[].release" {} \;
  find $1/data_stream -type f -path "*fields*" -name "ecs.yml" -exec yq e -i "unique_by(.name)" {} \;
  yq e -i 'del .release' manifest.yml
  yq e -i "with(select(.license != null); .conditions.elastic.subscription = .license) | del .license" manifest.yml
  ```

2. Next the `ecs-update` tool is run as follows:
  ```
  go run github.com/andrewkroh/go-examples/ecs-update@main -format-version=3.0.0 -fix-dotted-yaml-keys -add-owner-type -owner elastic/obs-cloudnative-monitoring packages/*
  ```
  
 Any remaining issues were fixed manually. Even though most of the changes here were generated by tooling, **this still needs careful review**  (hide whitespace changes to make your life easier). 

There should be _no functional changes_ to any of the packages here. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Review by gizas — APPROVED** [review]

File: packages/istio/manifest.yml [hunk 1]
```diff
@@ -1,18 +1,19 @@
-format_version: 1.0.0
+format_version: "3.0.0"
 name: istio
 title: Istio
 description: Collect logs and metrics from the service mesh Istio with Elastic Agent.
 type: integration
-version: 0.4.2
-release: beta
-license: basic
```
Comment (gizas) [inline packages/istio/manifest.yml:8]: This should not be removed

File: packages/prometheus/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,14 @@
 # newer versions go on top
+- version: 1.15.0
+  changes:
+    - description: Update the package format_version to 3.0.0.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/{{ PULL_REQUEST_NUMBER }}
```
Comment (gizas) [inline packages/prometheus/changelog.yml:6]: Also the version 1.15 here needs removal

File: packages/stan/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,14 @@
 # newer versions go on top
+- version: 1.5.0
+  changes:
+    - description: Update the package format_version to 3.0.0.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/{{ PULL_REQUEST_NUMBER }}
```
Comment (gizas) [inline packages/stan/changelog.yml:6]: Same here

**Review by gizas — CHANGES_REQUESTED** [review]
Summarising: 
- The license removal is a breaking change and must be fixed
- Also prometheus and packages have a wrong RP number

File: packages/istio/manifest.yml (hunk 1 above)
Comment (tommyers-elastic) [reply packages/istio/manifest.yml:8]: license is not in `conditions.elastic.subscription`

File: packages/docker/data_stream/event/fields/fields.yml [hunk 4]
```diff
@@ -1,11 +1,9 @@
 - name: docker.container.labels.*
   type: object
-  release: ga
```
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/docker/data_stream/event/fields/fields.yml:3]: yeh release tags are no longer allowed in field mappings

File: packages/istio/kibana/dashboard/istio-fed6f500-5941-11ed-bfb8-bbfe5b338339.json [hunk 5]
```diff
@@ -2551,16 +2551,6 @@
             "id": "metrics-*",
             "name": "controlGroup_d7ff30d3-5dcf-41d3-962d-682654add535:optionsListDataView",
             "type": "index-pattern"
-        },
-        {
```
Comment (tommyers-elastic) [inline packages/istio/kibana/dashboard/istio-fed6f500-5941-11ed-bfb8-bbfe5b338339.json:2555]: these references didn't point to anything. do you know what they are meant to do @gizas ?

Comment (tommyers-elastic) [reply packages/istio/kibana/dashboard/istio-fed6f500-5941-11ed-bfb8-bbfe5b338339.json:2555]: hm there is a new tags.yml for tagging saved objects, i wonder if these need migrating there.

Comment (tommyers-elastic) [reply packages/istio/kibana/dashboard/istio-fed6f500-5941-11ed-bfb8-bbfe5b338339.json:2555]: but they were flagged up as 'dangling references' by the validation - so i don' tthink they're doing anything

File: packages/kubernetes/manifest.yml [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
 description: Collect logs and metrics from Kubernetes clusters with Elastic Agent.
 type: integration
 categories:
   - observability
   - containers
   - kubernetes
 conditions:
-  kibana.version: "^8.10.0"
+  kibana:
+    version: ^8.10.1
```
Comment (tommyers-elastic) [inline packages/kubernetes/manifest.yml:13]: this is an intentional version bump - 8.10.0 doesn't exist

**Review by gizas — APPROVED** [review]
Looks fine with latest updates

**Pull request closed without merging**